src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -1291,7 +1291,7 @@ private JwtSecurityToken ValidateSignature(string token, JwtSecurityToken jwtTok
             if (string.IsNullOrEmpty(jwtToken.RawSignature))
             {
                 if (validationParameters.RequireSignedTokens)
-                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10504, token)));
+                    throw LogHelper.LogExceptionMessage(new SecurityTokenInvalidSignatureException(LogHelper.FormatInvariant(TokenLogMessages.IDX10504, JwtTokenUtilities.PrepareSecurityArtifact(token))));
```

Comment:
Could you use the JwtSecurityToken passed in the parameter?

Reply:
Would be nice if the method header described how these were different. Coming to this cold, it's not clear why both would exist.